examples/cube_tangents.rs

```
@@ -56,27 +35,32 @@ impl bevy_mikktspace::Geometry for Mesh {
         3
     }
 
-    fn position(&self, face: usize, vert: usize) -> Vec3 {
-        vertex(self, face, vert).position.into()
+    fn position(&self, face: usize, vert: usize) -> [f32; 3] {
+        vertex(self, face, vert).position
     }
 
-    fn normal(&self, face: usize, vert: usize) -> Vec3 {
-        vertex(self, face, vert).normal.into()
+    fn normal(&self, face: usize, vert: usize) -> [f32; 3] {
+        vertex(self, face, vert).normal
     }
 
     fn tex_coord(&self, face: usize, vert: usize) -> [f32; 2] {
-        vertex(self, face, vert).tex_coord.into()
+        vertex(self, face, vert).tex_coord
     }
 
-    fn set_tangent_encoded(&mut self, tangent: [f32; 4], face: usize, vert: usize) {
+    fn set_tangent(
+        &mut self,
+        tangent: Option<bevy_mikktspace::TangentSpace>,
```

This change allows both `set_tangent` methods to be combined, avoiding the somewhat ugly "implement one of these" pattern that was a holdover from the original C implementation.

Also note I've explicitly replaced this with an `Option`. The original C implementation will silently return a default value under certain circumstances, which I think is something worth highlighting to the user, as they may have a more sensible default. `TangentSpace` implements `Default`, so users can get the exact behaviour anyway.